tests/test_unit/test_roi/test_angles.py

```diff
@@ -0,0 +1,394 @@
+            {"boundary_only": True},
+            np.deg2rad(
+                np.array(
+                    [
+                        -135.0,
+                        0.0,
+                        90.0,
+                        -135.0,
+                        180.0,
+                    ]
+                )
+            ),
+            False,
+            id="[Allocentric] Force boundary calculation",
+        ),
+    ],
+)
+def test_ego_and_allocentric_angle_to_region(
+    push_into_range,
+    region: BaseRegionOfInterest,
+    fn_args: Iterable[Any],
+    fn_kwargs: dict[str, Any],
+    expected_output: xr.DataArray | Exception,
+    egocentric: bool,
+    request,
+) -> None:
+    """Test computation of the egocentric and allocentric angle.
+
+    Note, we only test functionality explicitly introduced in this method.
+    Input arguments that are just handed to other functions are not explicitly
+    tested here.
+
+    Specifically;
+
+    - ``in_degrees``,
+
+    The ``angle_rotates`` argument is tested in all cases (signs should be
+    reversed when toggling the argument).
+    """
+    if isinstance(region, str):
+        region = request.getfixturevalue(region)
+    if isinstance(expected_output, np.ndarray):
+        expected_output = xr.DataArray(data=expected_output, dims=["time"])
+
+    if egocentric:
+        which_method = "compute_egocentric_angle_to_nearest_point"
+        other_vector_name = "direction"
+    else:
+        which_method = "compute_allocentric_angle_to_nearest_point"
+        other_vector_name = "ref"
+    method = getattr(region, which_method)
+
+    if isinstance(expected_output, Exception):
+        with pytest.raises(
+            type(expected_output), match=re.escape(str(expected_output))
+        ):
+            method(*fn_args, **fn_kwargs)
+    else:
+        angles = method(*fn_args, **fn_kwargs)
+        xr.testing.assert_allclose(angles, expected_output)
+
+        # Check reversal of the angle convention,
+        # which should just reverse the sign of the angles,
+        # except for 180-degrees -> 180-degrees.
+        if fn_kwargs.get("in_degrees", False):
+            lower = -180.0
+            upper = 180.0
+        else:
+            lower = -np.pi
+            upper = np.pi
+        if (
+            fn_kwargs.get("angle_rotates", f"approach to {other_vector_name}")
+            == f"approach to {other_vector_name}"
+        ):
+            fn_kwargs["angle_rotates"] = f"{other_vector_name} to approach"
+        else:
+            fn_kwargs["angle_rotates"] = f"approach to {other_vector_name}"
+
+        reverse_angles = push_into_range(
+            method(*fn_args, **fn_kwargs), lower=lower, upper=upper
+        )
+        xr.testing.assert_allclose(
+            angles, push_into_range(-reverse_angles, lower=lower, upper=upper)
+        )
+
+
+@pytest.fixture
+def points_around_segment() -> xr.DataArray:
+    """Points around the segment_of_y_equals_x.
```

Comment:
```suggestion
    """Sample points on either side of the segment y=x.
```
maybe a bit more clear